homeassistant/components/octoprint/__init__.py

```diff
@@ -297,11 +298,17 @@ def update(self, sensor_type, end_point, group, tool=None):
         return response
 
 
-def get_value_from_json(json_dict, sensor_type, group, tool):
+def get_value_from_json(json_dict, sensor_type, path, tool):
     """Return the value for sensor_type from the JSON."""
+    paths = path.split(".", 1)
```

Comment:
This integration needs to use a standalone library published on PyPI for the device interface and data parsing code. We won't accept new features until that is done.